login_server/loginserver.py

```diff
@@ -351,9 +351,44 @@ def handle_http_request_message(self, msg):
                 'online_players_list': [p.display_name for p in self.players.values()],
                 'online_servers_list': online_game_servers_list
             }, sort_keys=True, indent=4))
+        elif msg.env['PATH_INFO'] == '/player':
+            if "REMOTE_ADDR" in msg.env:
+                self.logger.info('Served player stats request via HTTP to peer "' + msg.env["REMOTE_ADDR"] + '"')
+            else:
+                self.logger.info('Served player stats request via HTTP to Unknown peer')
+            
+            if "QUERY_STRING" in msg.env:
+                filtered_player_name = ''.join(filter(str.isalnum, msg.env['QUERY_STRING']))
+                player_data = False
+                if filtered_player_name in self.accounts:
+                    player_data = self.get_player_settings_data(filtered_player_name)
+                
+                if player_data:
+                    filtered_player_data = {
+                        "player_found": True,
+                        "clan_tag": player_data["clan_tag"],
+                        "player_name": filtered_player_name,
+                        "rank_xp": player_data["progression"]["rank_xp"]
+                    }
+                    msg.peer.send_response(json.dumps(filtered_player_data, sort_keys=True, indent=4))
+                else:
+                    msg.peer.send_response(json.dumps({
+                        'player_found': False
+                    }, sort_keys=True, indent=4))
+            else:
+                msg.peer.send_response(None)
         else:
             msg.peer.send_response(None)
 
+    def get_player_settings_data(self, player_name):
+        self.logger.info("Retrieved settings data for player: " + player_name)
```

Comment:
Do we care about this enough to log it?

Reply:
Good point. "Served player stats request..." is already logged, this log is triggered by the same set of functions so is probably unnecessary. I've removed this.